src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy

```diff
@@ -118,6 +119,11 @@ class ProtobufPlugin implements Plugin<Project> {
           // block. Only at this point the configuration has been finalized.
           project.protobuf.tools.registerTaskDependencies(project.protobuf.getGenerateProtoTasks().all())
         }
+
+        project.plugins.withType(IdeaPlugin) { IdeaPlugin ip ->
+            ip.model.module.sourceDirs += project.file("${project.protobuf.generatedFilesBaseDir}/main/java");
+            ip.model.module.generatedSourceDirs += project.file("${project.protobuf.generatedFilesBaseDir}/main/java");
+        }
```

Comment:
What's the difference between `ip.model.module.sourceDirs` and `ip.model.module.generatedSourceDirs`?
Why does the example for `build.gradle` only add to `sourceDirs` while here it adds to both?